### PR TITLE
Chef session logout issue on 401 error raised from chef servers API

### PR DESCRIPTION
**Pull request by sachin-msys**
### :nut_and_bolt: Description: What code changed, and why?
For module like automate infra view we are communicating with chef servers API's that is external API's in that case we don't want session gets expired on getting 401 error, instead of that we need to show user a error notification so that user can take a preventive action for error so we have introduced a `skip-interceptor` header for external API's to ignore chef session logout loop in auth interceptor service on getting 401 error from server.

### :chains: Related Resources
Fixes: #4137 , #4180 
### :+1: Definition of Done
Shown a authentication error notification on getting 401 error raised from chef servers API's please refer attached screenshot for the same. 

### :athletic_shoe: How to Build and Test the Change
Call infra proxy API which gives 401 error.
### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
chef-server-401

### CODE REVIEW

File: components/automate-ui/src/app/entities/cookbooks/cookbook-details.requests.ts [hunk 1]
```diff
@@ -13,8 +14,9 @@ export class CookbookDetailsRequests {
   public getCookbookDetails(
     server_id: string, org_id: string, cookbook_name: string, cookbook_version: string):
     Observable<CookbookDetails> {
+    const headers = new HttpHeaders().set(InterceptorSkipHeader, '');
     return this.http.get<RespCookbookDetails>(
-      `${env.infra_proxy_url}/servers/${server_id}/orgs/${org_id}/cookbooks/${cookbook_name}/${cookbook_version}`)
+      `${env.infra_proxy_url}/servers/${server_id}/orgs/${org_id}/cookbooks/${cookbook_name}/${cookbook_version}`, { headers })
```
Comment (sachin-msys) [inline components/automate-ui/src/app/entities/cookbooks/cookbook-details.requests.ts:19]: Introduced a special interceptor skip ` header ` in all cookbook requests.

File: components/automate-ui/src/app/modules/infra-proxy/cookbooks/cookbooks.component.ts [hunk 2]
```diff
@@ -19,10 +19,11 @@ import {
   styleUrls: ['./cookbooks.component.scss']
 })
 
-export class CookbooksComponent implements OnInit {
+export class CookbooksComponent implements OnInit, OnDestroy {
   @Input() serverId: string;
   @Input() orgId: string;
 
+  private isDestroyed = new Subject<boolean>();
```
Comment (scottopherson) [inline components/automate-ui/src/app/modules/infra-proxy/cookbooks/cookbooks.component.ts:26]: We're still not 100% consistent everywhere yet, but if we can append this with a `$` to visually indicate that it's an observable/subscribable value that'd be great! `isDestroyed$`

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html [hunk 3]
```diff
@@ -9,7 +9,7 @@
       </chef-thead>
       <chef-tbody>
         <chef-tr *ngFor="let dataBag of dataBags">
-          <chef-td><a [routerLink]="['/infrastructure','chef-servers', serverId, 'organizations', orgId, 'data_bags', dataBag.name]">{{ dataBag.name }}</a></chef-td>
+          <chef-td><a [routerLink]="['/infrastructure','chef-servers', serverId, 'organizations', orgId, 'data-bags', dataBag.name]">{{ dataBag.name }}</a></chef-td>
```
[the remark this answers is not in the document]
Comment (scottopherson) [reply components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html:12]: Nice thanks 👍

File: components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts [hunk 4]
```diff
@@ -15,6 +15,8 @@ import { ChefSessionService } from 'app/services/chef-session/chef-session.servi
 import * as selectors from 'app/services/projects-filter/projects-filter.selectors';
 import { ProjectsFilterOption } from '../projects-filter/projects-filter.reducer';
 
+export const InterceptorSkipHeader = 'Skip-Interceptor';
+
```
Comment (scottopherson) [inline components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts:19]: Good name choice and location for this constant 👍

File: components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts [hunk 5]
```diff
@@ -55,6 +57,12 @@ export class HttpClientAuthInterceptor implements HttpInterceptor {
         //           Right now, we're throwing the ID token across the internet for telemetry,
         //           where it's not needed. It would be nice to _not_ do that.
         headers = headers.set('Authorization', `Bearer ${id_token}`);
+        // Check the interceptor skip header in API request
+        // To avoid session logout if 401 raised from external API
+        if (request.headers.has(InterceptorSkipHeader)) {
```
[the remark this answers is not in the document]
Comment (scottopherson) [reply components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts:62]: 👍

**Pull request merged**